caddyhttp/proxy/upstream_test.go

```diff
@@ -448,3 +450,44 @@ func TestHealthCheckPort(t *testing.T) {
 	})
 
 }
+
+func TestHealthCheckContentString(t *testing.T) {
+	testString := "good"
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "blablabla%sblablabla", testString)
+		r.Body.Close()
+	}))
+
+	_, port, err := net.SplitHostPort(server.Listener.Addr().String())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer server.Close()
+
+	tests := []struct {
+		config string
+	}{
+		{"proxy / localhost:8080 {\n health_check \n health_check_port " + port +
+			" \n health_check_contentstring " + testString + "\n}"},
+		{"proxy / localhost:8080 {\n health_check / health_check_port " + port + " / health_check_contentstring " + testString + "\n}"},
+	}
+	for i, test := range tests {
+		_, err := NewStaticUpstreams(caddyfile.NewDispenser("Testfile", strings.NewReader(test.config)), "")
+		if err == nil {
+			t.Errorf("Test %d accepted invalid config", i)
+		}
+		resp, err := http.Get("http://localhost:" + port + "/")
+		if err != nil {
+			t.Errorf("Test %d error response: %s", i, err.Error())
+		}
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Errorf("Test %d reading body: %s", i, err.Error())
+		}
+		resp.Body.Close()
+		if !bytes.Contains(body, []byte(testString)) {
```

Comment:
What about testing the failure case, where the body does _not_ contain the "good" string?